convert to C

rofl0r opened the pull request.

rofl0r committed:
add proper Makefile

rofl0r committed:
convert the 2 only sources using C++ to C

it's ridiculous to add dependency on a C++ compiler and the complexity
of a mixed build for the sake of using a single class and new/delete
operators.

mckaygerhard commented:
This is a good one @kallisti5  !

mckaygerhard commented:
Usin standar well know makefile is better rather than a scons dependency that pull to install more things in my linux

rofl0r committed:
always print usage information when unknown argument passed

i tried like 5 minutes to get a list of command line options using
the regular --help, -? and what have you, but didn't figure out
the magic incantation "-h" on my own.

rofl0r committed:
show all available options in help output

rofl0r committed:
sound: clean up use of "silent"

rofl0r committed:
command line parser: don't crash on empty val

for parameters requiring an argument when none was passed.

rofl0r committed:
fix abysmall rendering performance

the 24bit, surface-based, directly-using-opengl code resulted in
huexpress running at 100% cpu load at about 30% native speed on
my machine. using SDL2's hardware-accelerated texture rendering
primitives instead (with 32bit textures), it now runs at 20%
cpu use at full speed.

rofl0r committed:
fix order of ini file vs command line parsing

this makes it possible to override settings from the ini file
from the command line.

rofl0r committed:
Log: make it possible to log to stdout instead of file

this will also be used if the logfile fails to open.

rofl0r commented:
added some commits to make rendering performance 15x faster, and fix the command line parser.

kallisti5 commented:
I'm not really interested in moving away from SCons.  Lots of upstream packages  / recipes are written around us using SCons.  Could the makefile change be opened into a separate PR to discuss?

Everything else looks fine.

rofl0r committed:
update scons build for recent changes

rofl0r committed:
INSTALL: add note that the Makefile isn't officially supported

rofl0r commented:
@kallisti5 i updated the SCons scripts and added a note to INSTALL that the makefile isn't supported, so imo it can be kept as a bonus but users that prefer scons know that it'll work and that's what's supposed to be used.
does this sound reasonable ?

rofl0r committed:
gfx.h: remove unused vars

rofl0r committed:
implement fullscreen toggle

rofl0r committed:
support the common ALT-enter combo for fullscreen toggle

additionally to F9, which was used in the past, although it probably
never worked anyway, at least since the SDL2 move.

rofl0r committed:
fix CRC calculation for unzipped ROMs

rofl0r committed:
CartLoad(): don't crash when fopen() fails

this happened when loading an iso from a .zip while no system card
roms were present in the paths searched.

rofl0r committed:
main: don't crash when PCEngine_LoadFile() fails

PCEngine_Run() will cause accesses to uninitialized structures
after PCEngine_LoadFile() returned error.

mckaygerhard commented:
> I'm not really interested in moving away from SCons. Lots of upstream packages / recipes are written around us using SCons. Could the makefile change be opened into a separate PR to discuss?
> 
> Everything else looks fine.

the problem with scons is that it depends of upstream and in some distros is not packaged.. also it changed with time to time.. currently i can build a piece of software with makefile with any version of.. not the same with scons or cmake if there's some change across versions

rofl0r commented:
> I'm not really interested in moving away from SCons

also this PR doesn't remove or deprecate scons, it simply adds the Makefile as an option - and since it's now documented as unofficial you won't have to maintain it.

rofl0r committed:
osd_sdl_gfx.c: fix compile with gcc10

gcc10 defaults to -fno-common and this results in a link error.

this patch also happens to be in freebsd's hugo tree:
https://cgit.freebsd.org/ports/tree/emulators/hugo/files/patch-osd__sdl__gfx.c

rofl0r committed:
osd_keyboard: fix crash displaying joystick name

src/osd_keyboard.c:450:29: warning:
passing argument 1 of 'SDL_JoystickName'
makes pointer from integer without a cast [-Wint-conversion]

mckaygerhard commented:
@rofl0r  the makefile works like a baby.. ..  about compilation it compiles in any machine.. i was using older debian wheeze

i tested and found a minimal detail.. when any rom started.. there's a peak of sound at start tht almost break my ears (with hearphones) i dont know if there's a inheric detail or a consecuence, 


![imagen](https://user-images.githubusercontent.com/1571724/162276960-8c71014a-2d57-4f3c-9bbe-9e72245ada16.png)

rofl0r commented:
> when any rom started.. there's a peak of sound at start tht almost break my ears

i experienced this too when i started hacking on huexpress (so it seems to be an old issue)

but it disappeared for me after [b17df90](https://github.com/kallisti5/huexpress/pull/16/commits/b17df906da5a65c7170562bab3f7bc424c4dd3a0) because before that commit huexpress was only running at like 20% speed so the sound was heavily distorted.
